### ISSUE TITLE
Implement __eq__ dunder method in graph classes to check for isomorphism

### ISSUE THREAD

**Issue by harryscholes** [networkx/networkx#2765]
Implementing the `__eq__` dunder method would permit graph isomorphism checking using the builtin equality operator, e.g.`G1 == G2`:

```
import networkx as nx

class DerivedGraph(nx.Graph):
    def __init__(self):
        super().__init__()
        
    def __eq__(self, other):
        return nx.is_isomorphic(self, other)
```
Test case:
```
G1 = DerivedGraph()
G1.add_path([0, 1, 2])

G2 = DerivedGraph()
G2.add_path([2, 1, 0])

assert G1 == G2, "Graphs are not isomorphic"
```
Thoughts? Happy to do this myself if I get the go ahead.

**Comment by hagberg** [comment]
Yeah, we could do that.  I know we have discussed this before somewhere - if I remember right the primary concern is that for larger graphs the isomorphism algorithm might have very long run times.   In that case it is not really appropriate for an equality test - or at least might be surprising to many users.

**Comment by harryscholes** [comment]
Yeah graph isomorphism has (quasi)polynomial time complexity, so it will be hard on larger graphs. This would also true for a user checking isomorphism with `is_isomorphic(G1, G2)` though. I guess we could add a checkpoint that raises a warning if the sizes of `G1` and `G2` are above a certain threshold (that could be overridden by the user). Also, I think it's generally a good idea to implement dunder methods to improve the UX of Python.
